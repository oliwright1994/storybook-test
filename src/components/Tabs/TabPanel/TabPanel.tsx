import { TabPanel as ChakraTabPanel, TabPanelProps as IChakraTabPanel } from '@chakra-ui/core'
import React, { FC } from 'react'

const TabPanel: FC<IChakraTabPanel> = ({ children, isSelected }) => {
  return <ChakraTabPanel isSelected={isSelected}>{children}</ChakraTabPanel>
}

export default TabPanel
