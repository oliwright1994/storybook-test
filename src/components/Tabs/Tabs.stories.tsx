import createStory from '@bright/storybook/createStory'
import React from 'react'

import Tabs from '.'

createStory('COMPONENTS | Tabs', [
  {
    label: 'Tabs',
    jsx: () => {
      return (
        <Tabs>
          <Tabs.TabList>
            <Tabs.Tab>Tab1</Tabs.Tab>
            <Tabs.Tab>Tab2</Tabs.Tab>
            <Tabs.Tab>Tab2</Tabs.Tab>
          </Tabs.TabList>
          <Tabs.TabPanels>
            <Tabs.TabPanel>Panel One</Tabs.TabPanel>
            <Tabs.TabPanel>Panel Two</Tabs.TabPanel>
            <Tabs.TabPanel>Panel Three</Tabs.TabPanel>
          </Tabs.TabPanels>
        </Tabs>
      )
    },
  },
])
