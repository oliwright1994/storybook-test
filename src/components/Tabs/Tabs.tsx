import { ITabs as IChakraTabs, Tabs as ChakraTabs } from '@chakra-ui/core'
import React, { FC } from 'react'

import Tab from './Tab'
import TabList from './TabList'
import TabPanel from './TabPanel'
import TabPanels from './TabPanels'

interface ITabs extends FC<IChakraTabs> {
  Tab: typeof Tab
  TabList: typeof TabList
  TabPanel: typeof TabPanel
  TabPanels: typeof TabPanels
}

const Tabs: ITabs = ({ children }) => {
  return <ChakraTabs  flexGrow={1}>{children}</ChakraTabs>
}

Tabs.Tab = Tab
Tabs.TabList = TabList
Tabs.TabPanel = TabPanel
Tabs.TabPanels = TabPanels

export default Tabs
