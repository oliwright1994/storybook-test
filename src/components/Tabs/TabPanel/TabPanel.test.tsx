import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Tabs from '../Tabs'
import TabPanel from '.'

describe('<TabPanel />', () => {
  const defaultProps = {
    id: '1',
    children: 'panel text',
    isSelected: true,
  }

  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(
      <Tabs>
        <Tabs.TabPanels>
          <TabPanel {...defaultProps} />
        </Tabs.TabPanels>
      </Tabs>
    )

    expect(container).toMatchSnapshot()
  })

  it('should render children', () => {
    const { container } = renderWithBrightTheme(
      <Tabs>
        <Tabs.TabPanels>
          <TabPanel {...defaultProps} />
        </Tabs.TabPanels>
      </Tabs>
    )

    expect(container).toHaveTextContent(defaultProps.children)
  })

  it('should not have hidden attribute when selected', () => {
    const { getByText } = renderWithBrightTheme(
      <Tabs>
        <Tabs.TabPanels>
          <TabPanel {...defaultProps} />
          <TabPanel id="2">panel test</TabPanel>
        </Tabs.TabPanels>
      </Tabs>
    )

    expect(getByText(defaultProps.children)).not.toHaveAttribute('hidden')
  })
})
