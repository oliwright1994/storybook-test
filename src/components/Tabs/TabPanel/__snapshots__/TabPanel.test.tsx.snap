// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPanel /> should render as expected 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-0"
      tabindex="-1"
    >
      <div
        aria-labelledby="tab:undefined"
        class="css-1jj07fx"
        id="panel:undefined"
        role="tabpanel"
        tabindex="-1"
      >
        panel text
      </div>
    </div>
  </div>
</div>
`;
